src/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -0,0 +1,326 @@
+            Interop.AppleCrypto.PAL_TlsProtocolId minProtocolId;
+            Interop.AppleCrypto.PAL_TlsProtocolId maxProtocolId;
+
+            switch (protocols & SupportedProtocols)
+            {
+                case SslProtocols.None:
+                    // REVIEW: Give this a message.
+                    throw new PlatformNotSupportedException();
+                case SslProtocols.Tls:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls10;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls10;
+                    break;
+                case SslProtocols.Tls11:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls11;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls11;
+                    break;
+                case SslProtocols.Tls12:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls12;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls12;
+                    break;
+                case SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls10;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls12;
+                    break;
+                case SslProtocols.Tls11 | SslProtocols.Tls12:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls11;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls12;
+                    break;
+                case SslProtocols.Tls | SslProtocols.Tls11:
+                    minProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls10;
+                    maxProtocolId = Interop.AppleCrypto.PAL_TlsProtocolId.Tls11;
+                    break;
+                default:
+                    // REVIEW: Give this a message ("contiguous protocol versions", type of thing).
+                    throw new PlatformNotSupportedException();
+            }
+
+            Interop.AppleCrypto.SslSetMinProtocolVersion(sslContext, minProtocolId);
+            Interop.AppleCrypto.SslSetMaxProtocolVersion(sslContext, maxProtocolId);
+        }
+
+        private static void SetCertificate(SafeSslHandle sslContext, X509Certificate2 certificate)
+        {
+            Debug.Assert(sslContext != null, "sslContext != null");
+            Debug.Assert(certificate != null, "certificate != null");
+            Debug.Assert(certificate.HasPrivateKey, "certificate.HasPrivateKey");
+
+            X509Chain chain = TLSCertificateExtensions.BuildNewChain(
+                certificate,
+                includeClientApplicationPolicy: false);
+
+            X509ChainElementCollection elements = chain.ChainElements;
+
+            // We need to leave off the EE (first) and root (last) certificate from the intermediates.
+            X509Certificate2[] intermediateCerts = elements.Count < 3 ?
```

Comment:
If 1, it's just the first cert.
If 2, it's a cert directly from a root.

In both 1 and 2 we've removed the items we care about by being Array.Empty.  At 3 we have things left over after we remove up to two.

And, really, it's because we're about to do `Count - 2` as an array size.